std/string.d

```diff
@@ -5316,9 +5320,13 @@ if (isSomeString!S)
 
 
 /***********************************************
+ * $(RED This function is deprecated and will be removed May 2018.)
+ * Please use the functions in $(MREF std, regex) and $(MREF std, algorithm)
+ * instead.
+ *
  * See if character c is in the intersection of the patterns.
  */
-
+deprecated("The std.string pattern functions are obsolete and will be removed May 2018. Use std.regex instead")
```

Comment:
It would be good to use the deprecated comment, s.t. this can be grepped for easily, e.g.

```
 // Explicitly undocumented. It will be removed in May 2018. @@@DEPRECATED_2018-05@@@
```

CC @jmdavis